src/Glpi/Form/AnswersHandler/AnswersHandler.php

```diff
@@ -74,6 +76,35 @@ public static function getInstance(): AnswersHandler
         return self::$instance;
     }
 
+    /**
+     * Check if the given answers are valid for the given form
+     *
+     * @param Form|Section $questions_container The form or section to check
+     * @param array $answers The answers to check
+     * @return ValidationResult The validation result
+     */
+    public function validateAnswers(
+        Form|Section $questions_container,
+        array $answers
+    ): ValidationResult {
+        $result = new ValidationResult();
+
+        // Check if mandatory questions are answered
+        $mandatory_questions = array_filter(
+            $questions_container->getQuestions(),
+            static fn($question) => $question->fields['is_mandatory']
+        );
+
+        foreach ($mandatory_questions as $question) {
+            // Check if the question is not answered (empty or not set)
+            if (empty($answers[$question->getID()])) {
+                $result->addError($question, __('This field is mandatory'));
+            }
+        }
```

Comment:
Does it take into account questions that are not visible ?

The original JS code only sent visible inputs to avoid errors on questions that may be hidden by a condition (we don't want to validate it in this case).

Reply:
One last remark for this point, from the client code I understand that you only send answers for the current section of the form.

This might create issue with the visibility engine as it is not impossible that a question from the current section depends on value from the previous sections.

In this case, its visibility would not be computed correctly here.